### DISCUSSION TITLE
More frequent data updates?

### DISCUSSION THREAD

**Discussion by hogsterbogster** [tomasmcguinness/homeassistant-mixergy#27] [General]
Hi there,

Firstly, thanks for making this integration! It's one less app I have to use separately now which is great! 😁

However, whilst the commands for setting the tank charge state get transmitted immediately to the tank (I can literally hear the relay click as soon as I've sent a heat command from HA), it can take a minute or two for the electrical power reading to update which, unless I'm mistaken, is the only way to tell that the tank is heating at the moment?

Is it possible to make this update more frequently, like once every 5 seconds perhaps?

Many thanks for your help!

David

**Comment by tomasmcguinness** [comment]
It polls the Mixergy API every 30 seconds. 

You're welcome to edit this in the sensor.py file under the integration. 

```
 # Create a coordinator to fetch data from the Mixergy API.
    coordinator = DataUpdateCoordinator(hass, _LOGGER, name="sensor", update_method = async_update_data, update_interval = timedelta(seconds=30))
```

You just need to change the timedelta to whatever you wish.

**Reply by hogsterbogster** [reply]
Amazing, thanks so much Tomas! 🙏

**Reply by hogsterbogster** [reply]
@tomasmcguinness - I never got round to changing the polling rate, but I've become aware of the function in Home Assistant to update an entity, which if I've understood it correctly, should force refresh that particular entity and not the whole integration.

This makes more sense than having the API being polled every few seconds when it's not doing anything for 99.9% of the day!

However, I tried implementing this as follows to update the electric heat power sensor reading straight after sending a heating request, but it didn't seem to make any difference.

![Screenshot 2024-03-02 at 22 07 51](https://github.com/tomasmcguinness/homeassistant-mixergy/assets/149208922/0e32f17d-a84f-466a-b7e6-1acfa34d9c89)

Is this the correct way to implement this, or does this integration not support updating specific entities / sensor readings?

It would be great if it could update 1 or 2 seconds after being sent a heat command.

Is there a way to send a single poll request directly from HA in the same way that a service is called? 

Many thanks for your help!

David

**Reply by tomasmcguinness** [reply]
Since adding my Kitchen control panel, I've realised that my integration remains out of date for a minute, so I do need to think of a way to update it almost immediately after, or to at least set a value (like target level) so you can see what's happening. 

I don't know about the support of updating a single entity, since that requires a request to the API, so it would update every other entity too.

**Reply by goldendel** [reply]
Hi Tomas,

Thanks for creating this great integration!  I'm trying to do something similar to David. I've created an automation to set my mixergy target charge % to 100% when I get an additonal cheap period while charging my EV.  However, there doesn't seem to be an action for stopping a boost via HA when the descretionary cheap period ends. Is this something already exposed via the API?

Thanks for your help,

Tim

**Reply by hogsterbogster** [reply]
@goldendel — I’ve just added a button that sets the charge level to 0% which acts as a cancel button, see the last row in the screenshot below:

![IMG_4104](https://github.com/tomasmcguinness/homeassistant-mixergy/assets/149208922/66bcd606-864b-4798-9ac0-33b7394e7e8a)

Maybe that will do what you’re after?

Cheers!

David

PS. One occasion where this doesn’t work is when trying to stop a scheduled boost … then you can’t even stop the boost in the Mixergy app, you have to turn the tank off via the button on the tank control board!

**Reply by goldendel** [reply]
@hogsterbogster That's really helpful.  I've implemented an automation to set the target charge to 0% and tested it with a bonus charge slot this evening.  The automation worked just as I wanted!

Thanks David.

Tim

PS How did you add the 3 drops in the card?

**Comment by hogsterbogster** [comment]
> @hogsterbogster That's really helpful. I've implemented an automation to set the target charge to 0% and tested it with a bonus charge slot this evening. The automation worked just as I wanted!
> 
> Thanks David.
> 
> Tim
> 
> PS How did you add the 3 drops in the card?

You're welcome Tim! 😀

The 3 drops is an emoji! 💦

![Screenshot 2024-04-04 at 22 37 04](https://github.com/tomasmcguinness/homeassistant-mixergy/assets/149208922/d3ef85f2-d129-4642-a7a7-8a47e6083dca)

Cheers!

David
